### PR TITLE
[Feature] remove EVM: add call contract logic for CitaExecutive:exec

### PR DESCRIPTION
**Pull request by leeyr338**
- add create contract logic for CitaExecutive:exec.

### CODE REVIEW

File: cita-executor/core/src/cita_executive.rs [hunk 1]
```diff
@@ -33,23 +34,26 @@ const AMEND_KV_H256: u32 = 3;
...
     env_info: EnvInfo,
     economical_model: EconomicalModel,
 }
 
-impl<B: DB + 'static> CitaExecutive<B> {
+impl<'a, B: DB + 'static> CitaExecutive<'a, B> {
     pub fn new(
         block_provider: Arc<BlockDataProvider>,
         state_provider: State<B>,
+        native_factory: &'a NativeFactory,
```
Comment (ouwenkg) [inline cita-executor/core/src/cita_executive.rs]: change to `native_provider`?

File: cita-executor/core/src/cita_executive.rs [hunk 2]
```diff
@@ -154,11 +158,7 @@ impl<B: DB + 'static> CitaExecutive<B> {
                     data: t.data.clone(),
                 };
 
-                self.create(
-                    self.block_provider.clone(),
-                    self.state_provider.clone(),
-                    &params,
-                )
+                self.call_evm(&params)
```
Comment (ouwenkg) [inline cita-executor/core/src/cita_executive.rs]: why  rename this? I think `create` may be better.

Comment (leeyr338) [reply cita-executor/core/src/cita_executive.rs]: I want to reuse the code with `call` function.

File: cita-executor/core/src/cita_executive.rs (hunk 1 above)
Comment (leeyr338) [reply cita-executor/core/src/cita_executive.rs]: If you carefully taste, `Factory` and `provider` are different, I think it is more reasonable to use `Factory` here.

Comment (ouwenkg) [reply cita-executor/core/src/cita_executive.rs]: OK，not a problem.

**Pull request merged**